### PR TITLE
[CoreGraphics] Add Support for DotNet Attributes

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

File: src/CoreGraphics/CGColorSpace.cs [hunk 1]
```diff
@@ -395,20 +403,52 @@ public byte[] GetColorTable ()
...
+#endif
 		extern static /* CGColorSpaceRef */ IntPtr CGColorSpaceCreateWithICCProfile (/* CFDataRef */ IntPtr data);
 
 		[DllImport (Constants.CoreGraphicsLibrary)]
+#if !NET
 		[iOS (10,0)][Mac (10,12)][Watch (3,0)][TV (10,0)]
+#else
+		[SupportedOSPlatform ("ios10.0")]
+		[SupportedOSPlatform ("tvos10.0")]
+		[SupportedOSPlatform ("macos10.12")]
```
Comment (spouliot) [inline src/CoreGraphics/CGColorSpace.cs:431]: min is iOS/tvOS 10.0 
mac min is 10.14

File: src/CoreGraphics/CGColorSpace.cs [hunk 2]
```diff
@@ -395,20 +403,52 @@ public byte[] GetColorTable ()
...
 		[DllImport (Constants.CoreGraphicsLibrary)]
+#if !NET
 		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'CreateIDCCData' instead.")]
+#else
+		[UnsupportedOSPlatform ("ios11.0")]
+		[UnsupportedOSPlatform ("tvos11.0")]
+		[UnsupportedOSPlatform ("macos10.13")]
```
Comment (spouliot) [inline src/CoreGraphics/CGColorSpace.cs:414]: below 10.14 min

File: src/CoreGraphics/CGColorSpace.cs [hunk 3]
```diff
@@ -395,20 +403,52 @@ public byte[] GetColorTable ()
...
 
+#if !NET
 		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
 		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'CreateIDCCData' instead.")]
+#else
+		[UnsupportedOSPlatform ("ios11.0")]
+		[UnsupportedOSPlatform ("tvos11.0")]
+		[UnsupportedOSPlatform ("macos10.13")]
```
Comment (spouliot) [inline src/CoreGraphics/CGColorSpace.cs:443]: below 10.14 min

File: src/CoreGraphics/CGColorSpace.cs [hunk 4]
```diff
@@ -462,19 +502,43 @@ public static CGColorSpace CreateICCProfile (nfloat[] range, CGDataProvider prof
...
 		[DllImport (Constants.CoreGraphicsLibrary)]
+#if !NET
 		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'GetICCData' instead." )]
+#else
+		[UnsupportedOSPlatform ("ios11.0")]
+		[UnsupportedOSPlatform ("tvos11.0")]
+		[UnsupportedOSPlatform ("macos10.13")]
```
Comment (spouliot) [inline src/CoreGraphics/CGColorSpace.cs:513]: below 10.14 min

File: src/CoreGraphics/CGColorSpace.cs [hunk 5]
```diff
@@ -462,19 +502,43 @@ public static CGColorSpace CreateICCProfile (nfloat[] range, CGDataProvider prof
...
 		[iOS (7,0)] // note: pre-release docs/headers says iOS6 and later, available on OSX since 10.5
-#endif
 		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'GetICCData' instead." )]
 		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'GetICCData' instead." )]
+#else
+		[UnsupportedOSPlatform ("ios11.0")]
+		[UnsupportedOSPlatform ("tvos11.0")]
+		[UnsupportedOSPlatform ("macos10.13")]
```
Comment (spouliot) [inline src/CoreGraphics/CGColorSpace.cs:533]: below 10.14 min

File: src/CoreGraphics/CGEvent.cs [hunk 6]
```diff
@@ -437,22 +437,30 @@ public static void PostToPSN (CGEvent evt, IntPtr processSerialNumber)
 			return result;
 		}
 
+#if !NET
 		[Mac (11,0)]
+#endif
```
Comment (spouliot) [inline src/CoreGraphics/CGEvent.cs:442]: missing `[SupportedOSPlatform ("macos11.0")]`

File: src/CoreGraphics/CGEvent.cs [hunk 7]
```diff
@@ -437,22 +437,30 @@ public static void PostToPSN (CGEvent evt, IntPtr processSerialNumber)
...
 		[DllImport (Constants.ApplicationServicesCoreGraphicsLibrary, EntryPoint="CGPreflightListenEventAccess")]
 		[return: MarshalAs (UnmanagedType.I1)]
 		public static extern bool PreflightListenEventAccess ();
 
+#if !NET
 		[Mac (11,0)]
```
Comment (spouliot) [inline src/CoreGraphics/CGEvent.cs:448]: missing `[SupportedOSPlatform ("macos11.0")]`

File: src/CoreGraphics/CGEvent.cs [hunk 8]
```diff
@@ -437,22 +437,30 @@ public static void PostToPSN (CGEvent evt, IntPtr processSerialNumber)
...
+#endif
 		[DllImport (Constants.ApplicationServicesCoreGraphicsLibrary, EntryPoint="CGRequestListenEventAccess")]
 		[return: MarshalAs (UnmanagedType.I1)]
 		public static extern bool RequestListenEventAccess ();
 
+#if !NET
 		[Mac (11,0)]
```
Comment (spouliot) [inline src/CoreGraphics/CGEvent.cs:455]: missing `[SupportedOSPlatform ("macos11.0")]`

File: src/CoreGraphics/CGEvent.cs [hunk 9]
```diff
@@ -437,22 +437,30 @@ public static void PostToPSN (CGEvent evt, IntPtr processSerialNumber)
...
+#endif
 		[DllImport (Constants.ApplicationServicesCoreGraphicsLibrary, EntryPoint="CGPreflightPostEventAccess")]
 		[return: MarshalAs (UnmanagedType.I1)]
 		public static extern bool PreflightPostEventAccess ();
 
+#if !NET
 		[Mac (11,0)]
```
Comment (spouliot) [inline src/CoreGraphics/CGEvent.cs:462]: missing `[SupportedOSPlatform ("macos11.0")]`

**Review by spouliot — CHANGES_REQUESTED** [review]

File: src/CoreGraphics/CGColorSpace.cs (hunk 2 above)
Comment (tj-devel709) [reply src/CoreGraphics/CGColorSpace.cs:414]: Some of these are for Unsupported though. Should I remove these for the private ones maybe or do all of these need to be removed?

File: src/CoreGraphics/CGEvent.cs (hunk 7 above)
Comment (tj-devel709) [reply src/CoreGraphics/CGEvent.cs:448]: Whoops! Good catch!

**Review by spouliot — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**